document/document_test.go

```diff
@@ -357,10 +360,13 @@ func TestNewFromStruct(t *testing.T) {
 	q := 5
 	u.Q = &q
 	u.R = new(group)
+	u.S = u.R
 	u.T = []int{1, 2, 3}
 	u.V = []*int{&q}
 	u.W = []user{u}
 	u.X = []interface{}{1, "foo"}
+	u.Y = [3]int{1, 2, 3}
```

Comment:
Same here, could you explain why you are changing these?

Reply:
same as above.